### PR TITLE
Group data sharing

### PR DESCRIPTION
**Pull request by ianmcorvidae**
Depends on cyverse-de/common-swagger-api#83

Doesn't currently include analysis sharing. I'll do that next.

### CODE REVIEW

File: src/terrain/services/sharing.clj [hunk 1]
```diff
@@ -172,12 +172,25 @@
...
+        provided-subject (:subject share-unshare)]
+    (or provided-user (get-user-from-subject provided-subject))))
+
 (defn- share-with-user
   "Forwards share requests to data-info from the user and list of paths and permissions in the given
    share map, sending any success notifications to the users involved, and any error notifications
    to the current user."
   [share]
-  (let [user (:user share)
+  (let [user (translate-user-for-irods share)
```
Comment (ianmcorvidae) [inline src/terrain/services/sharing.clj:193]: I figure that if for some reason something passes neither of user/subject, this'll just spit out a nil and that'll throw an error when it tries to pass it down the chain. I could also be more explicit about validation if desired though.

**Review by slr71 — APPROVED** [review]
Looks good to me! 👍

**Comment by ianmcorvidae** [comment]
Thanks for the review! I think with a release happening tomorrow, I'll hold off on merging this for right now. We don't have the UI stuff in place anyway, so the only user-facing effect it could have would be breaking stuff if I did something wrong.

**Review by psarando — APPROVED** [review]
LGTM 👍

**Pull request merged**